cmd/client-snapshot.go

```diff
@@ -0,0 +1,582 @@
+	if err != nil {
+		return nil, err
+	}
+	if b == nil {
+		return nil, probe.NewError(BucketDoesNotExist{Bucket: bucket})
+	}
+
+	u := s.PathURL.Clone()
+	u.Path = path.Join("/", b.Name)
+
+	// TODO: Include more information
+	return &ClientContent{
+		URL:  u,
+		Type: os.ModeDir,
+		Err:  nil,
+	}, nil
+}
+
+// Stat - get metadata from path.
+func (s *snapClient) Stat(ctx context.Context, _ StatOptions) (content *ClientContent, err *probe.Error) {
+	bucket, object := s.url2BucketAndObject()
+
+	if bucket == "" {
+		return nil, probe.NewError(BucketNameEmpty{})
+	}
+
+	if object == "" {
+		return s.statBucket(ctx, bucket)
+	}
+
+	b, err := s.dec.FindBucket(bucket)
+	if err != nil {
+		return nil, err
+	}
+	if b == nil {
+		return nil, probe.NewError(BucketDoesNotExist{Bucket: bucket})
+	}
+
+	object = strings.TrimSuffix(object, "/")
+
+	filter := func(entry *SnapshotEntry) filterAction {
+		if entry.IsDeleteMarker {
+			return filterSkipEntry
+		}
+		if strings.HasPrefix(entry.Key, object+"/") {
+			*entry = SnapshotEntry{Key: object + "/"}
+			return filterAbort
+		}
+		if entry.Key != object {
+			return filterSkipEntry
+		}
+		return filterAbort
+	}
+
+	contentCh := make(chan *ClientContent, 2)
+	s.getBucketContents(ctx, *b, contentCh, filter)
+	close(contentCh)
+
+	content = <-contentCh
+	if content == nil {
+		return nil, probe.NewError(ObjectMissing{})
+	}
+
+	if content.Err != nil {
+		return nil, content.Err
+	}
+
+	return content, nil
+}
+
+func (s *snapClient) AddUserAgent(_, _ string) {
+}
+
+// Get Object Tags
+func (s *snapClient) GetTags(ctx context.Context) (map[string]string, *probe.Error) {
+	return nil, probe.NewError(APINotImplemented{
+		API:     "GetObjectTagging",
+		APIType: "snapshot",
+	})
+}
+
+// Set Object tags
+func (s *snapClient) SetTags(ctx context.Context, tags string) *probe.Error {
+	return probe.NewError(APINotImplemented{
+		API:     "SetObjectTagging",
+		APIType: "snapshot",
+	})
+}
+
+// Delete object tags
+func (s *snapClient) DeleteTags(ctx context.Context) *probe.Error {
+	return probe.NewError(APINotImplemented{
+		API:     "DeleteObjectTagging",
+		APIType: "snapshot",
+	})
+}
+
+// Get lifecycle configuration for a given bucket, not implemented.
+func (s *snapClient) GetLifecycle(ctx context.Context) (*lifecycle.Configuration, *probe.Error) {
+	return nil, probe.NewError(APINotImplemented{
+		API:     "GetLifecycle",
+		APIType: "snapshot",
+	})
+}
+
+// Set lifecycle configuration for a given bucket, not implemented.
+func (s *snapClient) SetLifecycle(ctx context.Context, config *lifecycle.Configuration) *probe.Error {
+	return probe.NewError(APINotImplemented{
+		API:     "SetLifecycle",
+		APIType: "snapshot",
+	})
+}
+
+// Versioning operations
+
+// Not implemeneted
+func (s *snapClient) GetVersioning(ctx context.Context) (minio.BucketVersioningConfiguration, *probe.Error) {
+	return minio.BucketVersioningConfiguration{}, probe.NewError(APINotImplemented{
+		API:     "GetLifecycle",
```

Comment:
```suggestion
		API:     "GetVersioning",
```
Couldn't this be supported?